### PR TITLE
feat: Analyzing Reddit Comments Sentiment using MistralAI and LangGraph

### PR DESCRIPTION
**Pull request by norbinsh**
# Cookbook Pull Request

## Description

Added a new cookbook that utilizes MistralAI's large LLM with LangGraph to perform sentiment analysis on Reddit comments. 

...

## Type of Change

What type of PR is it?

- [X] New Cookbook
  - [X] Notebook File
    - [X] Does it work on google colab?
  - [X] Markdown File
- [ ] Cookbook Update
  - [ ] Code Refactoring
  - [ ] Bug Fix
- [ ] README.md Update
___
- [ ] Other (please describe):

## Cookbook Checklist:

- [X] My code is easy to read and well structured.
- [X] I've tagged the versions of any dependency required.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] My changes generate no new warnings or errors.
___
- [ ] My changes do not concern the cookbooks.

## README.md Checklist

- [X] I've added my cookbook to the table.
___
- [ ] My changes do not concern the README file.

## Additional Context

*Add any other context or screenshots about the feature request here.*

TL;DR

Added a new cookbook for analyzing Reddit comments' sentiment using MistralAI and LangGraph.

...

### CODE REVIEW

**Comment by norbinsh** [comment]
@sophiamyang hopefully i followed everything correctly, happy to tweak based on feedback.
Thanks!
